manager/manager/typeresolver.go

```diff
@@ -105,17 +105,19 @@ func (tr *typeResolver) ResolveTypes(config *common.Configuration, imports []*co
 			return nil, resolverError(config, fmt.Errorf("Failed to understand download url for %s: %v", r.Type, err))
 		}
 		if !existing[r.Type] {
-			f := &fetchUnit{}
-			for _, u := range urls {
-				if len(u) > 0 {
-					f.urls = append(f.urls, fetchableURL{urlRegistry, u})
-					// Add to existing map so it is not fetched multiple times.
-					existing[r.Type] = true
```

Comment:
Hoisting this line up before the for loop wouldn't work, since we don't want to set existing[r.Type] to true until we've tested it and done some work if we've found it to be false.

I think the problem here is with the description in the PR. This change doesn't fix a bug. Not sure what I was thinking. What it does is prevent us from repeatedly falling through in the case where there are no urls. Moving it doesn't change the behavior, but it makes the intent explicit, instead of relying on the fall through. Really just a matter of hygiene, since there are 3 tests involved in the fall through case, and someone making changes here might not notice a behavioral change.

The if len(urls) > 0 check is also not strictly necessary, since without it, we would fall through the for _, u := range urls block and then through the if len(f.urls) > 0 block, doing nothing. Again, it seems cleaner to make the intent explicit than to rely on fall through behavior.

Frankly, this change isn't worth the time we're spending on it. I just made the changes for the sake of code hygiene as I was reading through this code. However, since you aren't comfortable with them, and we've already spent too much time on it, I'll just close the PR.